pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -1038,3 +1042,18 @@ func RemovePipelineFinalizer(ctx context.Context, pr *pipelinev1beta1.PipelineRu
 	}
 	return reconcile.Result{}, nil
 }
+
+// This is to remove any '#xxx' fragment from a URI so that git clone commands don't need seperate adjustment
+func modifyURLFragment(log logr.Logger, paramValues *[]pipelinev1beta1.Param) {
```

Comment:
I would modify this before it ends up in the parameter array. Basically instead of using `db.Spec.ScmInfo.SCMURL` around line 563 when setting the parameter, store this in a variable and strip it from the variable.

Reply:
Sure. Out of curiosity what is `PipelineParamChainsGitUrl` used for?